### ISSUE TITLE
Prow github status has giant blank space in the middle, looks weird

### ISSUE THREAD

**Issue by dobesv** [kubernetes/test-infra#28923]
**What happened**:

When you look at the status of prow jobs in a github PR it has a whole bunch of white space after a job name, then "BaseSHA: xxxx...."

![image](https://user-images.githubusercontent.com/327833/222852210-c93b47b1-709e-436a-bfb6-52bead69f607.png)

![image](https://user-images.githubusercontent.com/327833/222852640-6e62836e-f587-499f-b9a7-e479615ea5d5.png)

This looks weird, and seems like a bug.

Looking at the code, it seems like it was kind of done intentionally but I don't think it is having the intended effect of showing this in a "stable position" as you can see in the screenshot.

```
// ContextDescriptionWithBaseSha is used by the GitHub reporting to store the baseSHA of a context
// in the status context description. Tide will read this if present using the BaseSHAFromContextDescription
// func. Storing the baseSHA in the status context allows us to store job results pretty much forever,
// instead of having to rerun everything after sinker cleaned up the ProwJobs.
func ContextDescriptionWithBaseSha(humanReadable, baseSHA string) string {
	var suffix string
	if baseSHA != "" {
		suffix = contextDescriptionBaseSHADelimiter + baseSHA
		// Leftpad the baseSHA suffix so its shown at a stable position on the right side in the GitHub UI.
		// The GitHub UI will also trim it on the right side and replace some part of it with '...'. The
		// API always returns the full string.
		if len(humanReadable+suffix) < contextDescriptionMaxLen {
			for i := 0; i < contextDescriptionMaxLen-len(humanReadable+suffix); i++ {
				// This looks like a standard space but is U+2001, because GitHub seems to deduplicate normal
				// spaces in their frontend.
				suffix = " " + suffix
			}
		}
	}
	return truncate(humanReadable, contextDescriptionMaxLen-len(suffix)) + suffix
}
```

**Expected behavior**

I think it would just look better if there was no left-padding of the BaseSHA but instead it was just added to the status string.

**How to reproduce it (as minimally and precisely as possible)**:

It's visible on all the pull request in this repo

**Please provide links to example occurrences, if any**:

https://github.com/kubernetes/test-infra/pull/28922

**Anything else we need to know?**:

It seems like a bit of a hack putting this stuff into the status message.  I do wonder if there's a better place to stash this information, for example it could be added onto the target URL as an unused query param or hash instead, that way it doesn't affect the user experience as much.

**Comment by BenTheElder** [comment]
> It seems like a bit of a hack putting this stuff into the status message. I do wonder if there's a better place to stash this information, for example it could be added onto the target URL as an unused query param or hash instead, that way it doesn't affect the user experience as much.

IIRC yes, and it was previously hidden in the UI doing this hack (I think). @matthyx worked on this.

>  I do wonder if there's a better place to stash this information, for example it could be added onto the target URL as an unused query param or hash instead, that way it doesn't affect the user experience as much.

Interesting thought, not sure it's safe or if that information shows up in the APIs being used.

**Comment by matthyx** [comment]
I wrote the plugin, but not the UI changes (I wouldn't be able to). How can I help here?

**Comment by BenTheElder** [comment]
I'm not sure how we should approach this.

**Comment by amirrmonfared** [comment]
> I'm not sure how we should approach this.
@BenTheElder I would be able to take a look and investigate it. if you have any suggestions please share.

**Comment by dobesv** [comment]
I don't mind working on a PR if there's an agreed-on approach.

**Comment by amirrmonfared** [comment]
> Is there some reason not to just remove all the blank space in there? Might be the simplest solution

I agree on, the simplest and best way is this since having it in URL param cannot be the reliable way. also if we little bit beautify this by adding `—` can also bring better user experience

**Comment by BenTheElder** [comment]
Sorry I just can't prioritize this at the moment. The K8s Infra funding issue is taking full priority for the immediate future. We have other people active in prow though.

**Comment by petr-muller** [comment]
I must admit I prefer the current state of things where the clutter is at least partially hidden. I'd like to see a stronger majority than the current two voices in favor of the change.

**Comment by amirrmonfared** [comment]
> I must admit I prefer the current state of things where the clutter is at least partially hidden. I'd like to see a stronger majority than the current two voices in favor of the change.

Would be better to hear more opinions, but in my point of view since we cannot completely hide it so just including it is making a better user experience.
if you can please mention some people who could help us with this.

**Comment by petr-muller** [comment]
> we cannot completely hide it so just including it is making a better user experience. 

I fail to see how this implication is true. IMO it's quite the opposite: "we cannot completely hide it so let's just move it out of the way as much as possible".

**Comment by amirrmonfared** [comment]
> I fail to see how this implication is true. IMO it's quite the opposite: "we cannot completely hide it so let's just move it out of the way as much as possible".

IIRC this is already done by the previous implementation.

**Comment by petr-muller** [comment]
Yeah, which is why I want to current implementation to stay ;)
